services-geocoding/src/main/java/com/mapbox/api/geocoding/v6/V6StructuredInputQuery.java

```diff
@@ -0,0 +1,165 @@
+  }
+
+  /**
+   * This builder is used to create a new instance that holds structured input request options.
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * The number associated with the house.
+     *
+     * @param addressNumber structured input component.
+     * @return this builder for chaining options together
+     */
+    public abstract Builder addressNumber(@NonNull String addressNumber);
+
+    /**
+     * The name of the street in the address.
+     *
+     * @param street structured input component.
+     * @return this builder for chaining options together
+     */
+    public abstract Builder street(@NonNull String street);
+
+    /**
+     * In some countries like Japan, the block is a component in the address.
+     *
+     * @param block structured input component.
+     * @return this builder for chaining options together
+     */
+    public abstract Builder block(@NonNull String block);
+
+    /**
+     * Typically these are cities, villages, municipalities, etc.
+     * They are usually features used in postal addressing, and are suitable for display in ambient
+     * end-user applications where current-location context is needed
+     * (for example, in weather displays).
+     *
+     * @param place structured input component.
+     * @return this builder for chaining options together
+     */
+    public abstract Builder place(@NonNull String place);
+
+    /**
+     * Top-level sub-national administrative features, such as states in the United States
+     * or provinces in Canada or China.
+     *
+     * @param region structured input component.
+     * @return this builder for chaining options together
+     */
+    public abstract Builder region(@NonNull String region);
+
+    /**
+     * Postal codes used in country-specific national addressing systems.
+     *
+     * @param postcode structured input component.
+     * @return this builder for chaining options together
+     */
+    public abstract Builder postcode(@NonNull String postcode);
+
+    /**
+     * Official sub-city features present in countries where such an additional administrative
+     * layer is used in postal addressing, or where such features are commonly referred to in local
+     * parlance. Examples include city districts in Brazil and Chile and arrondissements in France.
+     *
+     * @param locality structured input component.
+     * @return this builder for chaining options together
+     */
+    public abstract Builder locality(@NonNull String locality);
+
+    /**
+     * Colloquial sub-city features often referred to in local parlance. Unlike locality features,
+     * these typically lack official status and may lack universally agreed-upon boundaries.
+     *
+     * @param neighborhood structured input component.
+     * @return this builder for chaining options together
+     */
+    public abstract Builder neighborhood(@NonNull String neighborhood);
```

Comment:
This class doesn't have `public static TypeAdapter<V6StructuredInputQuery> typeAdapter(Gson gson)` function, thus AutoValue generates only private constructor and this class can't be deserialised from Gson (and we don't expect it to be serialised/deserialised since it's part of options, not response).

And even if this had available constructor, the more important is get function nullability, which is marked with @Nullable
```
  @SerializedName("neighborhood")
  @Nullable
  abstract String neighborhood();
```

I think it doesn't make sense for customers to set null as it's a default value. But we can mark parameters in Builder as Nullable as well, I don't have strong opinion on this.